### PR TITLE
♻️Refactor Dashboard and Tasklist View

### LINKED ISSUE #821: 🎨 Refactor dashboard task list view
#### Describe your issue

Currently, the task list is defined twice. 

Once in the `src/modules/dashboard/dashboard.html` and in `src/modules/task-list/task-list.html`

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by alexanderkasten**
:tada: **Thanks for submitting this Pull Request**

Please provide a list including your changes.

To help us to create a meaningful Changelog, we agreed 
to use an emoji-based commit-style. Use the [Emojis](#emojis)
we provided below.

Fill out the following, provide reference to issues
and this PR- we will use this as merge commit message-body
and the title of this PR as commit-headline:

---

**Changes:**

1. Lets the dashboard view uses the task-list.
2. Adds a new component that uses the task-list for processmodels and correlations.
3. Removes xml-beautifier from aurelia config

**Issues:**

Closes #821 

PR: #PullRequest

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/modules/task-list/task-list.ts [hunk 1]
```diff
@@ -89,24 +73,25 @@ export class TaskList {
     } else {
       this._getUserTasks = this._getAllUserTasks;
     }
-    this.updateUserTasks();
+    this._updateUserTasks();
   }
 
   public attached(): void {
     if (!this._getUserTasks) {
```
Comment (SteffenKn) [inline src/modules/task-list/task-list.ts]: You could refactor this line as well by moving the business rule to an own variable

File: src/modules/task-list-container/task-list-container.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+interface ITaskListRouteParameters {
+  processModelId?: string;
+  correlationId?: string;
+}
+
+@inject('NotificationService', Router, 'ManagementApiClientService', 'AuthenticationService')
+export class TaskListContainer {
+
+  public showTaskList: boolean = false;
+  public tasklist: TaskList;
```
Comment (SteffenKn) [inline src/modules/task-list-container/task-list-container.ts]: taskList

File: src/modules/task-list-container/task-list-container.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    this._authenticationService = authenticationService;
+  }
+
+  public async canActivate(): Promise<boolean> {
+    const managementContext: ManagementContext = this._getManagementContext();
+
+    const hasClaimsForTaskList: boolean = await this._hasClaimsForTaskList(managementContext);
+
+    if (!hasClaimsForTaskList) {
+      this._notificationService.showNotification(NotificationType.ERROR, 'You don\'t have the permission to use the dashboard features.');
```
Comment (SteffenKn) [inline src/modules/task-list-container/task-list-container.ts]: Maybe something like:
`'You don\'t have the permission to use the inspect features.'`
is easier to understand.

File: src/modules/task-list/task-list.ts [hunk 4]
```diff
@@ -254,4 +239,20 @@ export class TaskList {
...
+    try {
+      this._userTasks = await this._getUserTasks();
+      this.successfullyRequested = true;
+    } catch (error) {
+      if (isError(error, UnauthorizedError)) {
+        this._notificationService.showNotification(NotificationType.ERROR, 'You don\'t have permission to view the task list.');
+        this._router.navigateToRoute('start-page');
+      } else {
+        this._notificationService.showNotification(NotificationType.ERROR, `Error receiving task list: ${error.message}`);
+      }
```
Comment (SteffenKn) [inline src/modules/task-list/task-list.ts]: You may add some spacing to the whole function to improve readability

File: src/modules/task-list-container/task-list-container.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    this._managementApiService = managementApiService;
+    this._authenticationService = authenticationService;
+  }
+
+  public async canActivate(): Promise<boolean> {
+    const managementContext: ManagementContext = this._getManagementContext();
+
+    const hasClaimsForTaskList: boolean = await this._hasClaimsForTaskList(managementContext);
+
+    if (!hasClaimsForTaskList) {
```
Comment (SteffenKn) [inline src/modules/task-list-container/task-list-container.ts]: Get rid of the exclamation mark in the if

```
    const hasNoClaimsForTaskList: boolean = !(await this._hasClaimsForTaskList(managementContext));

     if (hasNoClaimsForTaskList) {
```

File: src/modules/task-list/task-list.ts [hunk 6]
```diff
@@ -89,24 +73,27 @@ export class TaskList {
     } else {
       this._getUserTasks = this._getAllUserTasks;
     }
-    this.updateUserTasks();
+    this._updateUserTasks();
```
Comment (SteffenKn) [inline src/modules/task-list/task-list.ts]: Add some spacing

```
    }

    this._updateUserTasks();
```

**Comment by Vyperus** [comment]
Please solve merge conflicts

**Comment by alexanderkasten** [comment]
@Vyperus done

**Pull request merged**